atom/browser/web_dialog_helper.cc

```diff
@@ -82,13 +82,15 @@ class FileSelectHelper : public base::RefCounted<FileSelectHelper>,
  private:
   friend class base::RefCounted<FileSelectHelper>;
 
+  std::unique_ptr<net::DirectoryLister> lister_;
+
   ~FileSelectHelper() override {}
 
   void EnumerateDirectory(base::FilePath base_dir) {
-    auto* lister = new net::DirectoryLister(
+    lister_.reset(new net::DirectoryLister(
         base_dir, net::DirectoryLister::NO_SORT_RECURSIVE,
-        new atom::DirectoryListerHelper(base_dir, this));
-    lister->Start();
+        new atom::DirectoryListerHelper(base_dir, this)));
```

Comment:
It would also be good if you can eliminate `atom::DirectoryListerHelper` , its an unneeded redirection for representing a delegate. `FileSelectHelper` can very well represent itself as the delegate.